src/media/media.ts

```diff
@@ -115,30 +111,51 @@ export class SdkMedia extends (EventEmitter as { new(): StrictEventEmitter<Event
    *   video: false
    * }
    * ```
-   * @param mediaType media type to request permissions for (`'audio' | 'video'`)
+   *
+   * Note #2: if requesting `'both'`, it will always ask for both permision types
+   *   even if the first is denied. For example, if `audio` permissions are denied, this
+   *   function will still ask for `video` permissions. If _at least_ one media permission
+   *   was denied, it will throw the error. Exception is if `IMediaRequestOptions`'
+   *   `preserveMediaIfOneTypeFails` option is `true`, then it will only the error if _both_
+   *   media types fail. And then it will throw the audio denied error.
+   *
+   * Note #3: in some browsers, permissions are not always guaranteed even after
+   *   using this function to gain permissions. See [Firefox Behavior] for more details.
+   *   It is recommended to use the [permissions event](#permissions) to watch for changes
+   *   in permissions since permissions could be denied anytime `startMedia()` is called.
+   *
+   * @param mediaType media type to request permissions for (`'audio' | 'video' | 'both'`)
    * @param preserveMedia flag to return media after permissions pass
    * @param options optional, advanced options to request media with.
    *
    * @returns a promise either containing a `MediaStream` or `undefined`
    *   depending on the value of `preserveMedia`
    */
   async requestMediaPermissions (
-    mediaType: 'audio' | 'video',
+    mediaType: 'audio' | 'video' | 'both',
```

Comment:
So one thing that's kind of weird about this is if you request `both`, but don't provide a `audio` and/or `video` property, it won't be a part of the stream. Is there a better way we could handle this do you think?

Reply:
We will _always_ request the media type passed in as the first param regardless of what the `options` contains. On [line 151](https://github.com/MyPureCloud/genesys-cloud-webrtc-sdk/pull/583/files#diff-9cb49fb78c3beea80b0ffdf12a129e9131b8ad8b7d94af4743fb87db15d59554R151) we validate the option. If `false` or `undefined` is passed in, we will default to `true` (sdk default).